app/src/main/java/com/github/featuredetectandroid/utils/LuminanceArrayToBitmap.kt

```diff
@@ -0,0 +1,36 @@
+package com.github.featuredetectandroid.utils
+
+import android.graphics.Bitmap
+import android.util.Log
+import java.nio.IntBuffer
+
+private const val TAG = "LuminanceArrayToBitmap"
+
+private const val ALPHA_SHIFT = 24 // Alpha channel shift
+private const val RED_SHIFT = 16 // Red channel shift
+private const val GREEN_SHIFT = 8 // Green channel shift
+private const val BLUE_SHIFT = 0 // Blue channel shift
+
+private const val Y_SHIFT = 127 // Y-component (luminance) should be inverted and shifted
+
+private const val MAX_COLOR = 0xff
+private const val MAX_ALPHA = MAX_COLOR shl ALPHA_SHIFT
+
+fun luminanceArrayToBitmap(luminanceByteArray: ByteArray, width: Int, height: Int): Bitmap {
+    Log.i(TAG, "Converting ByteArray image of size $width x $height to a Bitmap.")
+
+    val buffer = IntBuffer.allocate(luminanceByteArray.size).apply {
+        for (byte in luminanceByteArray) {
+            val intByte = Y_SHIFT - byte.toInt()
+            put(
+                MAX_ALPHA or (intByte shl RED_SHIFT)
+                    or (intByte shl GREEN_SHIFT) or (intByte shl BLUE_SHIFT)
```

Comment:
[According to Android code style](https://developer.android.com/kotlin/style-guide#where_to_break), line break should come after `or`. Also, if I press `Crtl+Alt+L` Android Studio adds additional indents here.

Reply:
fixed